src/services/resources-getter.js

```diff
@@ -168,23 +150,14 @@ function ResourcesGetter(model, opts, params) {
       });
     }
 
-    return query;
-  }
-
-  function exec(query) {
-    return new P((resolve, reject) => {
-      if (params.sort) {
-        handleSortParam(query);
-      }
+    if (params.sort) {
+      handleSortParam(jsonQuery);
+    }
```

Comment:
Is it necessary to sort when we only count records? (same question for skip and limit)

Reply:
Moved inside the perform